File: src/main/java/komu/blunt/objects/CompoundProcedure.java
package komu.blunt.objects;

import komu.blunt.eval.Environment;

import static com.google.common.base.Preconditions.checkArgument;
import static com.google.common.base.Preconditions.checkNotNull;

public final class CompoundProcedure {
    
    public final int address;
    public final Environment env;

    public CompoundProcedure(int address, Environment env) {
        checkArgument(address >= 0);

        this.address = address;
        this.env = checkNotNull(env);
    }

    @Override
    public String toString() {
        return "<#CompoundProcedure " + address + ">";
    }
}

File: src/main/kotlin/komu/blunt/objects/CompoundProcedure.kt
package komu.blunt.objects

import komu.blunt.eval.Environment

class CompoundProcedure(val address: Int, val env: Environment) : Procedure {

    init {
        require(address >= 0) { "negative address: $address" }
    }

    override fun toString() = "<#CompoundProcedure $address>"
}
